### DISCUSSION TITLE
RealTime Wave Constructing

### DISCUSSION THREAD

**Discussion by Gsouzabd** [katspaugh/wavesurfer.js#3809] [Q&A]
Hi, is there any way to create a realtime wave usin record plugin?

For example:
![image](https://github.com/user-attachments/assets/964e96bb-37f6-4ef1-92e4-465e3193e9a1)

as long as the user is talking, it goes building.

Its like the 'scrollingWaveform' params, but we need to build from left to right in a specific time.

**Comment by katspaugh** [comment]
It should be possible but it’s not a feature that is offered out of the box.

**Reply by Gsouzabd** [reply]
Hello mate!
Can you indicate some params that i could use in v7?

**Reply by katspaugh** [reply]
Like I said, there are no readily available params that you can use but you could perhaps subscribe to the `record-data-available` event and accumulate the recorded data in an array, then use that as "peaks" of another wavesurfer instance. See https://wavesurfer.xyz/docs/types/plugins_record.RecordPluginEvents

**Reply by Gsouzabd** [reply]
@katspaugh  would you be willing to work on a task to help us building this feature?
If so, can you provide your email?

**Reply by katspaugh** [reply]
https://wavesurfer.xyz/contact-us/

**Reply by Gsouzabd** [reply]
I sent you an email. Please, check it out asap.
Tks!
